### ISSUE TITLE
Add to cart spining too long

### ISSUE THREAD

**Issue by tase09** [facebook/facebook-for-woocommerce#2827]
## 🔎 Isolate the bug

- [x] I have confirmed this occurs in the most recent version of WordPress, WooCommerce, and Facebook for WooCommerce.
- [x] I have confirmed this occurs when only WooCommerce and Facebook for WooCommerce are active and when using a default WordPress or WooCommerce theme.

## ✍️ Describe the bug
Recently from today we discovered on many woocommerce websites that the plugin is making add to cart button loading too hard when adding to cart

## 🚶‍♀️ Steps to reproduce
Add to cart

**Comment by WouterBossen** [comment]
Here for exactly this. turning off the plugin immediately fixes the problem.

**Comment by WouterBossen** [comment]
It also impacts the checkout page.

**Comment by martinstr** [comment]
Same here; also, clicking from the product listing into a single product takes forever when the Facebook for WooCommerce plugin is activated.

**Comment by gruberro** [comment]
We experienced  the same issue and had to disable either the plugin or disconnect the connection. The error seems to be related to the Facebook app managed by Automattic, as the app is no longer available and requests to FB are timing out.

An example error is:

```
2024-11-21T08:55:53+00:00 NOTICE Response
code: 400
message: Bad Request
body: {"error":{"message":"Cannot call API for app xxx on behalf of user xxx","type":"OAuthException","code":200,"fbtrace_id":"AZq0AEzSlTFR-xxx"}}
 CONTEXT: {"_legacy":true}
```

**Comment by WouterBossen** [comment]
> We experienced the same issue and had to disable either the plugin or disconnect the connection. The error seems to be related to the Facebook app managed by Automattic, as the app is no longer available and requests to FB are timing out.
> 
> An example error is:
> 
> ```
> 2024-11-21T08:55:53+00:00 NOTICE Response
> code: 400
> message: Bad Request
> body: {"error":{"message":"Cannot call API for app xxx on behalf of user xxx","type":"OAuthException","code":200,"fbtrace_id":"AZq0AEzSlTFR-xxx"}}
>  CONTEXT: {"_legacy":true}
> ```

Can confirm the issue is also happening when viewing uncached single product pages.

**Comment by Aironas47** [comment]
our company facing same issue. We had to disconect more than 50 ecommerce stores. Hope this problem will be resolved soon

**Comment by wahidnory** [comment]
Same issue here, solved directly after disabling the plugin.

**Comment by tase09** [comment]
> We experienced the same issue and had to disable either the plugin or disconnect the connection. The error seems to be related to the Facebook app managed by Automattic, as the app is no longer available and requests to FB are timing out.
> 
> An example error is:
> 
> ```
> 2024-11-21T08:55:53+00:00 NOTICE Response
> code: 400
> message: Bad Request
> body: {"error":{"message":"Cannot call API for app xxx on behalf of user xxx","type":"OAuthException","code":200,"fbtrace_id":"AZq0AEzSlTFR-xxx"}}
>  CONTEXT: {"_legacy":true}
> ```

so you are saying the app will be discountinued ? so we need to change also the feed of products? they will stop to sinc?

**Comment by WouterBossen** [comment]
> our company facing same issue. We had to disconect more than 50 ecommerce stores and turn of facebook plugin. Hope this problem will be resolved soon

Without the plugin though, events will not be handled anymore which could seriously impact your marketing strategy on facebook and instagram. That's why this is HUGE! Really hope there will be a fix for this asap.

**Comment by tase09** [comment]
so i found a temporary fix, u can keep enabled the plugin if u put the pixel in parallel with another plugin and it  stops messing with the website, so for the moment this is a good fix, events will not duplicate

**Comment by WouterBossen** [comment]
> so i found a temporary fix, u can keep enabled the plugin if u put the pixel in parallel with another plugin and it stops messing with the website, so for the moment this is a good fix, events will not duplicate

Could you elaborate on this? Am I correct in understanding that you mean inserting the plugin, say, through GTM and not through the plugin?

**Comment by Sebastian1005** [comment]
We have another issue which is related.

it seems the Facebook-APP from the plugin with CLIENT_ID facebook-for-woocommerce/includes/Handlers/Connection.php is deactivated in facebook

**Comment by gutobenn** [comment]
> On my end it looks like the problem has been fixed. Can anyone else confirm?

Yes, it's working again

**Comment by gruberro** [comment]
> On my end it looks like the problem has been fixed. Can anyone else confirm?

I can confirm as well!

**Comment by aquezada** [comment]
Me to!

**Comment by martinstr** [comment]
it works now
